Sources/NIO/BaseSocketChannel.swift

```diff
@@ -18,14 +18,16 @@ private struct SocketChannelLifecycleManager {
     // MARK: Types
     private enum State {
         case fresh
-        case registered
+        case lazilyRegistered // register() has been run but the selector doesn't know about it yet
+        case fullyRegistered // fully registered, ie. the selector knows about it
         case activated
         case closed
     }
 
     private enum Event {
         case activate
-        case register
+        case registerLazily
+        case registerFully
```

Comment:
Similarly, these names aren't necessarily right. Maybe `beginRegistration` and `completeRegistration`?